Maintenance(testing): follow up for benchmarking

sdwvit opened the pull request.

Commit:
- benchmark webserver now hosts styles too

Commit:
- add google fonts to bench html

Commit:
- change number of iterations and path to html file in tests

Commit:
- change loop to go max 50 iterations between refreshes
- change exit condition to not exit if minimum number of iterations wasn't reached
- simplify code a bit

Commit:
- move bench scripts to the scripts/test folder

Commit:
- Update file paths for JIRA, Slack, and comment files to be relative
 - Implement unique fingerprint generation for failed tests to avoid duplicates
 - Add functionality to reopen existing JIRA issues if found
 - Improve output messages for better clarity during issue creation

Commit:
- Update transition ID for reopening JIRA issues to 'TODO'
 - Modify automated message format for issue creation

sdwvit committed:
Merge remote-tracking branch 'origin/sdwvit/maintenance/testing/host-test-files-under-https' into sdwvit/maintenance/testing/host-test-files-under-https

sdwvit committed:
- Change loop condition in benchmarking.ts to use Math.max for iterations
 - Remove unused imports from cron.spec.ts and setData.spec.ts
 - Update webServer.js to use ES module syntax and improve path handling
 - Modify static file serving logic in webServer.js for healthcheck

sdwvit committed:
Merge branch 'latest' into sdwvit/maintenance/testing/host-test-files-under-https

# Conflicts:
#	testing/performance/benchmarking.ts
#	testing/performance/webServer.js

sdwvit committed:
- Update test result formatting to include Git Diff links conditionally.
 - Change default charts version to v12.0.0 across environments.
 - Add warning for fingerprint generation when no failed tests are found.

sdwvit committed:
- Replace hardcoded project key with a constant for better maintainability
 - Introduce constants for custom field fingerprints and track values
 - Update JQL query to use the project constant and filter for empty fixVersion
 - Set parent ticket ID for grid performance items in the JIRA issue body

sdwvit committed:
- Add IS_SUCCESS environment variable to determine test outcome
- Replace COLUMN_TODO_ID with BACKLOG and QA column mappings
- Introduce commonFetch helper for Jira API requests with error handling
- Update workflow condition to create Jira tickets based on test success
- Modify ticket creation logic to transition existing issues to QA on success

StephenCooper commented on testing/performance/webServer.js:
```diff
@@ -22,9 +27,15 @@ mkcert
-        app.use('/', express.static(staticTestFiles));
+        app.use('/', (req, res, next) => {
+            if (req.url === '/') {
+                res.send('OK', 200);
```
Comment explaining the requirement of this.

sdwvit committed:
- Update Jira column IDs and project id to RTI
 - Add /healthcheck route to web server for monitoring purposes
 - Modify Jira ticket script to use async/await for better flow control
 - Update Playwright config to use health check URL for web server

sdwvit committed:
- add comment

sdwvit committed:
- Added PERFORMANCE_CHAMP_USER_IDS for Jira assignee randomization
 - Updated fingerprint generation to use all tests instead of failed ones
 - Removed unused waitForMs import from playwright.utils

sdwvit committed:
- Allow github comment even if jira issue creation failed

sdwvit committed:
- add issue id to a comment for easier tracking

sdwvit committed:
- Update Playwright test command to use glob pattern for cron.spec.ts
 - Modify test case name to include 'Scheduled' for clarity

sdwvit committed:
- undo automation of reassigning issue after transitioning

sdwvit committed:
- update glob
